src/core_modules/capture-core/components/D2Form/D2Form.container.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+import { connect } from 'react-redux';
+import type { ComponentType } from 'react';
+import { compose } from 'redux';
+import { withStyles } from '@material-ui/core';
+import type { Props, PropsFromRedux, OwnProps } from './D2Form.types';
+import { withAsyncHandler } from './asyncHandlerHOC';
+import { styles, D2FormComponent } from './D2Form.component';
+
+const mapStateToProps = (state: ReduxState, ownProps: OwnProps): PropsFromRedux => {
+    const { forms } = state;
+
+    const isFormInReduxStore = !!forms[ownProps.id];
```

Comment:
padding this property inside the component to decide if we would like the form to be shown or not.